File: sdks/javascript/examples/hello-actor.ts
import { Actor, Message, MessageType, Capability } from '../src';

class HelloActor extends Actor {
    static get name() { return 'hello'; }

    async handle(sender: string, message: Message): Promise<Message | void> {
        if (message.type === MessageType.CUSTOM) {
            const name = message.payload.name || 'World';
            return Message.custom({ greeting: `Hello, ${name}!` });
        }
    }
}

const actor = new HelloActor({ name: 'hello' });
actor.start().catch(console.error);

File: sdks/python/examples/hello_actor.py
from typing import Optional

from aether_sdk import Actor, Message, MessageType


class HelloActor(Actor):
    @classmethod
    def name(cls) -> str:
        return "hello"

    async def handle_message(self, sender: str, message: Message) -> Optional[Message]:
        if message.payload.get("type") == "greet":
            name = message.payload.get("name", "World")
            return Message(
                type=MessageType.CUSTOM, payload={"greeting": f"Hello, {name}!"}
            )
        return None


if __name__ == "__main__":
    import asyncio

    async def main():
        actor = HelloActor()

        test_msg = Message(
            type=MessageType.CUSTOM, payload={"type": "greet", "name": "Aether"}
        )
        response = await actor.handle_message("test", test_msg)
        if response:
            print(f"Response: {response.payload}")

    asyncio.run(main())
